go/vt/vttablet/tabletserver/state_manager.go

```diff
@@ -99,6 +99,10 @@ type stateManager struct {
 	alsoAllow      []topodatapb.TabletType
 	reason         string
 	transitionErr  error
+	// requestsWaitCounter is the number of goroutines that are waiting for requests to be empty.
+	// If this value is greater than zero, then we have to ensure that we don't Add to the requests
+	// to avoid any panics in the wait.
+	requestsWaitCounter int
```

Comment:
nit: can use atomic counter

Reply:
should this be a bool. as we want only one `sm.requests.Wait()` to happen at a time.

Reply:
We have mutex protection, so atomic counter would be the same thing. Plus I think its better to be consitant with the struct. If we have a mutex to protect all internal state changes of a struct, we should use it.
I made it an int instead of a boolean because even though theoritically we should only be calling `sm.requests.Wait()` only once, its not explicitly enforced or written down. int just feels a safer option and more future proof.